packages/fast-css-editor-react/src/spacing/spacing.props.ts

```diff
@@ -0,0 +1,58 @@
+import { ManagedClasses } from "@microsoft/fast-jss-manager";
+import { CSSSpacingClassNameContract } from "./spacing.style";
+
+export enum SpacingType {
+    margin = "margin",
+    padding = "padding",
+}
+
+export enum SpacingKey {
+    marginTop = "marginTop",
+    marginLeft = "marginLeft",
+    marginRight = "marginRight",
+    marginBottom = "marginBottom",
+    paddingTop = "paddingTop",
+    paddingBottom = "paddingBottom",
+    paddingLeft = "paddingLeft",
+    paddingRight = "paddingRight",
+}
+
+export interface CSSSpacingState {
+    activeType: SpacingType;
+
+    hoverType: SpacingType | undefined;
+}
+
+export interface CSSSpacingUnhandledProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export interface CSSSpacingValues {
+    marginTop?: string;
+    marginLeft?: string;
+    marginBottom?: string;
+    marginRight?: string;
+    paddingTop?: string;
+    paddingBottom?: string;
+    paddingLeft?: string;
+    paddingRight?: string;
+}
+
+export interface CSSSpacingHandledProps
+    extends ManagedClasses<CSSSpacingClassNameContract>,
+        CSSSpacingValues {
+    /**
+     * The type of spacing margin/padding
+     */
+    spacingType?: SpacingType;
+
+    /**
+     * The spacing update callback
+     */
+    onSpacingUpdate?: (spacing: CSSSpacingValues) => void;
+
+    /**
+     * The spacing type update callback
+     */
+    onSpacingTypeUpdate?: (spacingType: SpacingType) => void;
```

Comment:
These will all get served to the editor component which aggregates them all, that is why they are retaining the additional "spacing" so that its clear when implementing editor eg.

```
<CSSEditor
    ...
    onSpacingTypeUpdate={this.handleSpacingTypeUpdate}
/>
```